packages/main/src/requests/request-helpers.ts

```diff
@@ -104,6 +106,31 @@ function assignRoleToPartsAndValidateSendMessageRequest(
   return functionContent;
 }
 
+export function formatCountTokensInput(
+  params: CountTokensRequest | string | Array<string | Part>,
+  model: string,
+): CountTokensRequestInternal {
+  let formattedRequest: CountTokensRequestInternal = {};
+  const containsGenerateContentRequest =
+    (params as CountTokensRequest).generateContentRequest !== undefined;
```

Comment:
Could use `!= null` which includes undefined, in case the user has decided to do weird things and null it out or something.

Reply:
Done.